components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -117,29 +112,19 @@ func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner even
 			}
 		}
 
-		// clean the application name segment in the event-type from none-alphanumeric characters
-		eventType, err := cleaner.Clean(eventType)
-		if err != nil {
-			return false, err
-		}
-
-		sub, subscribeErr := n.connection.Subscribe(eventType, callback)
+		natsSub, subscribeErr := n.connection.Subscribe(subject, callback)
 		if subscribeErr != nil {
 			n.log.Error(subscribeErr, "failed to create a Nats subscription")
 			return false, subscribeErr
 		}
-		n.subscriptions[fmt.Sprintf("%s.%s", namespacedName.String(), filter.EventType.Value)] = sub
+		n.subscriptions[getKey(sub, subject)] = natsSub
 	}
 	return false, nil
 }
 
 func (n *Nats) DeleteSubscription(subscription *eventingv1alpha1.Subscription) error {
-	subNsName := types.NamespacedName{
-		Namespace: subscription.Namespace,
-		Name:      subscription.Name,
-	}
 	for k, v := range n.subscriptions {
```

Comment:
Even if `k` and `v` for key and value are technically correct I would appreciate `key` and `subscription`.

Reply:
Done